src/Functions/FunctionsHashing.h

```diff
@@ -354,12 +365,22 @@ struct MurmurHash3Impl64
 struct MurmurHash3Impl128
 {
     static constexpr auto name = "murmurHash3_128";
-    enum { length = 16 };
 
-    static void apply(const char * begin, const size_t size, unsigned char * out_char_data)
+    using ReturnType = UInt128;
+
+    static UInt128 combineHashes(UInt128 h1, UInt128 h2)
     {
-        MurmurHash3_x64_128(begin, size, 0, out_char_data);
+        return apply(reinterpret_cast<const char *>(&h1), 16) ^ h2;
```

Comment:
Hm..that doesn't look like something similar to 64-bit version.

Reply:
Thank you for your review. Do you have any suggestions? IntHash64Impl is used in the 64-bit version.

Reply:
I think this should be implemented as `combine(x, y) = someHashFunction(x || y)` where `||` is "concatenation".
And probably we should change 32 and 64 bit version the same way.

Reply:
done.